test/ci-build.sh

```diff
@@ -26,7 +30,7 @@ export CC
 # Standard build
 for CC in gcc gcc-9 gcc-10 clang; do
     echo "=== Building with ${CC} ==="
-    mkdir build-${CC}; cd build-${CC}
+    mkdir build-${CC}; pushd build-${CC}
```

Comment:
You could probably also use a sub shell for the loop body, instead of manually keeping pushd and popd in sync?

(Oh, but I see that we are using pushd and popd already elsewhere in the script. So probably better not to change that in this PR.)

Reply:
All individual commits - just should be explained in the commit message.